File: new/atoms/is-encrypted.test.mjs
import { isEncrypted as __pure_isEncrypted } from './is-encrypted.mjs';
// צילום-מקומי + עטיפת-כריכה (מנוע-הטיהור v3)
const __d_isEncrypted_IS_ENCRYPTED_T = {
  k1: "object",
};
const isEncrypted = (...a) => __pure_isEncrypted(...a, ...Array(Math.max(0, 1 - a.length)).fill(undefined), __d_isEncrypted_IS_ENCRYPTED_T);
const C = [
  [{ $enc: 2, iter: 600000, data: '...' }, true],
  [{ $enc: 2 }, true],
  [{ $enc: 1 }, false],
  [{ $enc: '2' }, false],
  [null, false],
  ['{"$enc":2}', false],
  [{}, false],
];
let f = 0;
for (const [a, w] of C) {
  const g = isEncrypted(a);
  if (g !== w) { console.error(`✗ ${JSON.stringify(a)} ⇒ ${g} ≠ ${w}`); f = 1; }
}
if (f) process.exit(1);
console.log('✓ is-encrypted: 7 דוגמאות-חוזה — ירוק');

File: new/dart-maor/is-encrypted_test.dart
// בדיקת-חוזה (רתמת-זהב) · isEncrypted — מייבאת אך ורק את האטום-שלה (חוק-4).
// שבע דוגמאות-החוזה זהות ביט-אחר-ביט למקור-ה-JS new/atoms/is-encrypted.test.mjs:
//   1) {$enc:2, iter:600000, data:'...'}  ⇒ true   (מעטפת מלאה)
//   2) {$enc:2}                            ⇒ true   ($enc===2 מספיק)
//   3) {$enc:1}                            ⇒ false  (מספר שונה)
//   4) {$enc:'2'}                          ⇒ false  (מחרוזת ≠ מספר, strict)
//   5) null                                ⇒ false  (!!raw)
//   6) '{"$enc":2}'                        ⇒ false  (מחרוזת אינה object)
//   7) {}                                  ⇒ false  ($enc חסר → undefined/null ≠ 2)
// המרה: === של JS ⇒ == ב-Dart (String≠int בלי throw); `$enc` ⇒ מפתח r'$enc'.
// הרצה: dart run --enable-asserts new/dart-maor/is-encrypted_test.dart  ⇒ exit 0
import 'is-encrypted.dart';

void _ok(bool cond, String msg) {
  if (!cond) throw StateError('FAIL: $msg');
}

void main() {
  var n = 0;

  final cases = <List<dynamic>>[
    [{r'$enc': 2, 'iter': 600000, 'data': '...'}, true],
    [{r'$enc': 2}, true],
    [{r'$enc': 1}, false],
    [{r'$enc': '2'}, false],
    [null, false],
    ['{"\$enc":2}', false],
    [<String, dynamic>{}, false],
  ];

  for (final c in cases) {
    final input = c[0];
    final want = c[1] as bool;
    final got = isEncrypted(input);
    _ok(got == want, '$input ⇒ $got ≠ $want');
    n++;
  }

  // assert חי (חוק: --enable-asserts) — מוכיח שהמנגנון פעיל.
  assert(isEncrypted({r'$enc': 2}) == true, 'assert-live guard');
  assert(isEncrypted({r'$enc': '2'}) == false, 'assert-live guard strict');

  print('OK isEncrypted: $n asserts passed');
}
